cmd/metal-api/internal/headscale/client.go

```diff
@@ -119,6 +119,29 @@ func (h *HeadscaleClient) CreatePreAuthKey(namespace string, expiration time.Tim
 	return resp.PreAuthKey.Key, nil
 }
 
+func (h *HeadscaleClient) DescribeMachine(hostname, projectID string) (connected bool, err error) {
```

Comment:
LGTM, the only downside actually is that hostname is not guaranteed to be unique in a project. It is the case for our clusters but in the pure metal use case it is not. Should be at least documented.

Reply:
One idea would be to add a tag from the firewall during connecting with:

`tailscale up --advertise-tags tag:metal-stack.io/machineid=<id of the firewall> `

and then check if the machineid tag with exists

Reply:
Yes sure

Reply:
Situation is even more complicated then i initially thought. Headscale API refuses to return any tags if ACL policy isn't set. So tags are very tightly connected to ACLs.